drivers/hwinfo/hwinfo_handlers.c

```diff
@@ -14,3 +14,18 @@ ssize_t z_vrfy_hwinfo_get_device_id(u8_t *buffer, size_t length)
 	return z_impl_hwinfo_get_device_id((u8_t *)buffer, (size_t)length);
 }
 #include <syscalls/hwinfo_get_device_id_mrsh.c>
+
+ssize_t z_vrfy_hwinfo_get_reset_cause(u32_t * cause)
+{
+	Z_OOPS(Z_SYSCALL_MEMORY_WRITE(cause, sizeof(u32_t)));
+
+	return z_impl_hwinfo_get_reset_cause((u32_t *)cause);
```

Comment:
@andrewboie has already pointed the link to the documentation that explains how to proper validate parameters, take a look there. In essence you have to use a local copy  and then copy to user using `z_user_to_copy`.

Reply:
Thank you @ceolin. This system call writes the result to `cause`. We don't use the user passed value. This is similar behaviour as `hwinfo_get_device_id` so I followed the same validation. What am I missing?

Reply:
That one should be changed as well, take a look in https://docs.zephyrproject.org/latest/reference/usermode/syscalls.html#verification-memory-access-policies

Reply:
That documentation states that local copying should be done if we read the parameter value, since the user can modify that between validation and use. For this call we're not. We're only copying data out. The example discourages use of `Z_SYSCALL_MEMORY_WRITE` if we're going to read the data. We are not.

Having said that, I don't mind following your suggestion and changing it to use the local copy. I just want to make sure I am understanding this correctly, because I am new to Zephyr development. Thanks again!

Reply:
Hi,
The documentation might be unclear. But we don't want to pass any user pointers to implementation functions except for the documented exception of large buffers (which this isn't.)
If `cause` is an output-only parameter, pass a stack variable address to the implementation function, and z_user_to_copy() it afterwards.

Reply:
Thank you. I'll make the change.